lvhdev/lifeio-website
# Add survey pages

## [LIFE-100](https://lifeio.atlassian.net/browse/LIFE-100) Build Typeform Survey Pages

In order to track users who complete our Typeform survey (to reward them with points), we will provide each user with a personalized URL via Intercom. This URL will contain query params for the corresponding `name` and `email` hidden fields created on Typeform. (Intercom's messaging system has placeholders that act like string interpolation, so building the custom URL is easy - see pic below.)

Sample URLs:
Non-Metlife user: `http://life.io/survey/?name=Katelyn&email=katelyn@example.com`
Metlife user: `http://life.io/wellbeing-survey/?name=Jon&email=jon@example.com`

![screen shot 2016-03-22 at 1 39 11 pm](https://cloud.githubusercontent.com/assets/10477795/13961676/edf766c6-f033-11e5-8ec0-ed44514a7b9f.png)


This change addresses the need by:
- Adds two pages, `wellbeing-survey.html` for Metlife users, and `survey.html` for non-Metlife users
- Both pages use JavaScript to pull params from the URL and pass them to the embedded Typeform survey via string interpolation.

This workaround is necessary in order to use Typeform's hidden fields feature, which isn't already built in to their provided embedded survey code.

**Steps to test:**
- Pull down this branch, then [install](https://github.com/lvhdev/lifeio-website/blob/master/README.md) (if necessary) and load the website locally with `gulp serve`
- Copy/paste this URL and check that the survey greets you by name (feel free to use your own name): `http:/localhost:9000/survey/?name=Katelyn&email=katelyn@example.com`
- Repeat with `http://localhost:9000/wellbeing-survey/?name=Jon&email=jon@example.com`
- Basically, it just shouldn't greet you with the placeholder `Hi {{name}}!` because that means it's broken :broken_heart:
